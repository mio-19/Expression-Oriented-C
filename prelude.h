#define let auto
#define begin(body) ({body})
#define start(body) {return ({body});}
#define the(type,value) ({(type) ___EOC___temp___VARIABLE___1___=(value);___EOC___temp___VARIABLE___1___})
#define as(type,value) (type)(value)
static inline void mkvoid(void){}
