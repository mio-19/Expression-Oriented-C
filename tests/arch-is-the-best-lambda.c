#include <stdio.h>
#include <stdlib.h>
#include <prelude.h>
int main(void) start(
   let arch_best = lambda_returns(void)(puts("Arch is the best!"); mkvoid());
   void (*arch_best2)(int) = lambda_returns(void, int x)(printf("Arch is the best! %d\n", x);mkvoid());
   arch_best();
   arch_best2(9999);
   EXIT_SUCCESS)
