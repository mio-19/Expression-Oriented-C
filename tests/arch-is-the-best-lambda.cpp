#include <iostream>
#include <cstdlib>
#include <prelude.h>
int main () start(
   let arch_best = lambda_returns(void, int x)(std::cout << "Arch is the best! " << x << std::endl;);
   arch_best(9999);
   EXIT_SUCCESS)
