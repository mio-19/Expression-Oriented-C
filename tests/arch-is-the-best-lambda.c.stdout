Arch is the best!
Arch is the best! 9999
